### PR TITLE
MC-39996: Add support for Composer 2

### PR DESCRIPTION
**Pull request by Rrego6**
- Added composer2 support

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "magento/inventory-composer-installer",
-    "version": "1.1.0",
+    "version": "1.1.1",
```
Comment (fascinosum) [inline composer.json:3]: I would recommend to increase a minor version

File: src/Plugin.php [hunk 2]
```diff
@@ -89,4 +89,12 @@ private function getModuleDeployment(): InventoryModuleDeployment
         }
         return $this->moduleDeployment;
     }
+
+    public function deactivate(Composer $composer, IOInterface $io)
```
Comment (fascinosum) [inline src/Plugin.php:93]: please add `@inheritDoc`

**Review by fascinosum — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Rrego6) [reply composer.json]: Fixed

File: src/Plugin.php (hunk 2 above)
Comment (Rrego6) [reply src/Plugin.php:96]: Fixed

**Review by fascinosum — APPROVED** [review]

**Review by sidolov — APPROVED** [review]

**Comment by hostep** [comment]
Great!

But please next time take a look at the open PR's on the repo and try to acknowledge the community members who already put time and effort into this (#3, #4), thanks!

**Comment by fascinosum** [comment]
Hi @hostep,
sorry we missed them. There were a bunch of projects we needed to upgrade and test in order to have Composer 2 for Magento 2.4.2 in time.
This won't be the case if we regularly process PRs for this repo. @sidolov do we have a process in place?

**Pull request merged**